code/game/objects/items/power_cells.dm

```diff
@@ -13,6 +13,10 @@
 	var/maxcharge = 1000
 	var/rigged = 0		// true if rigged to explode
 	var/minor_fault = 0 //If not 100% reliable, it will build up faults.
+	var/self_recharge = FALSE // If true, the cell will recharge itself.
```

Comment:
nothing in this PR makes use of self_recharge

Reply:
Made the pulse magazine able to self_recharge.